File: java/json-transform/src/test/java/co/nlighten/jsontransform/functions/TransformerFunctionJsonPatchTest.java
package co.nlighten.jsontransform.functions;

import co.nlighten.jsontransform.BaseTest;
import org.junit.jupiter.api.Test;

public class TransformerFunctionJsonPatchTest extends BaseTest {

    @Test
    void add() {
        assertTransformation(fromJson("""
    {
        "a": {
            "b": "c"
        }
    }"""), fromJson("""
    {
        "$$jsonpatch": "$",
        "ops": [
            { "op":"add", "path":"/a/d", "value":"e" }
        ]
    }"""), fromJson("""
    {
        "a": {
            "b": "c",
            "d": "e"
        }
    }"""));

    }
}

File: javascript/json-transform/src/__tests__/functions/TransformerFunctionJsonPatch.test.ts
import { describe, test } from "vitest";
import { assertTransformation } from "../BaseTransformationTest";

describe("TransformerFunctionJsonPatch", () => {
  test("add", async () => {
    await assertTransformation(
      {
        a: {
          b: "c",
        },
      },
      {
        $$jsonpatch: "$",
        ops: [{ op: "add", path: "/a/d", value: "e" }],
      },
      {
        a: {
          b: "c",
          d: "e",
        },
      },
    );
  });
});
